PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/projects/SbtProjectGenerator.java

```diff
@@ -173,17 +176,38 @@ private String generateSbtFileContent() {
 
     private void importSbtProject(String root) {
         Project project = this.module.getProject();
+        final ProjectSystemId externalSystemId = new ProjectSystemId("SBT");
+        final ExternalSystemManager<?,?,?,?,?> manager = ExternalSystemApiUtil.getManager(externalSystemId);
+        if (manager == null) {
+            return;
+        }
 
-        ExternalSystemUtil.refreshProject(project,
-                SbtProjectSystem.Id(), root,
-                false, ProgressExecutionMode.IN_BACKGROUND_ASYNC);
+        ProjectImportProvider[] projectImportProviders = new ProjectImportProvider[1];
+        for (ProjectImportProvider provider : ProjectImportProvider.PROJECT_IMPORT_PROVIDER.getExtensions()) {
+            if (provider instanceof AbstractExternalProjectImportProvider
+                    && externalSystemId.equals(((AbstractExternalProjectImportProvider)provider).getExternalSystemId()))
+            {
+                projectImportProviders[0] = provider;
+                break;
+            }
+        }
 
-        SbtProjectSettings sbtProjectSettings = new SbtProjectSettings();
-        sbtProjectSettings.setExternalProjectPath(root);
+        if (projectImportProviders[0] == null) {
+            return;
+        }
 
-        SbtSystemSettings sbtSystemSettings = SbtSystemSettings.getInstance(project);
-        HashSet<SbtProjectSettings> projects = ContainerUtilRt.newHashSet(sbtSystemSettings.getLinkedProjectsSettings());
-        projects.add(sbtProjectSettings);
-        sbtSystemSettings.setLinkedProjectsSettings(projects);
+        final VirtualFile projectFile = VfsUtil.findFile(Paths.get(root + "/build.sbt"), true);

```

Comment:
I'm not sure whether the path delimiter will cause issues in different platforms, but `Paths.get(root, "build.sbt")` looks better for me.

Reply:
Thanks! To fix it!